### PR TITLE
replace Add/Remove buttons in ActionView with -/+ buttons per row

### PR DESCRIPTION
**Pull request by owenca**
1 row
multiple rows

### CODE REVIEW

File: sources/ActionView.cpp [hunk 1]
```diff
@@ -29,9 +30,30 @@ ActionView::ActionView(const char* name, BMessage* action, const int32& flags)
...
+	BSize size(width, width);
+
+	fAddButton->SetExplicitSize(size);
+	fRemoveButton->SetExplicitSize(size);
+
 	BLayoutBuilder::Group<>(this, B_HORIZONTAL, B_USE_DEFAULT_SPACING)
 		.Add(fActionField, 0)
 		.Add(fValueBox)
+		.Add(fRemoveButton, 0)
+		.Add(fAddButton, 0)
```
Comment (humdingerb) [inline sources/ActionView.cpp]: If you do this instead, there's no gap between the +/- buttons, which looks IMO a bit nicer:
```
	BLayoutBuilder::Group<>(this, B_HORIZONTAL, 0)
		.Add(fActionField, 0)
		.AddStrut(B_USE_ITEM_SPACING)
		.Add(fValueBox)
		.AddStrut(B_USE_ITEM_SPACING)
		.Add(fRemoveButton, 0)
		.Add(fAddButton, 0)
		.End();
```

Comment (owenca) [inline sources/ActionView.cpp]: Good idea. I use `B_USE_HALF_ITEM_SPACING` to make it tighter:
git diff

Comment (humdingerb) [inline sources/ActionView.cpp]: Yes HALF_ITEM tightens things up nicely.

I also realized, that there's a spacing issue if the action has only one pop-up menu, i.e. no text control, like "Delete". Add it as a second action and see that there's now a bigger gap in front of the +/- buttons.

This solves it:

```
	BLayoutBuilder::Group<>(this, B_HORIZONTAL, B_USE_HALF_ITEM_SPACING)
		.Add(fActionField, 0)
		.Add(fValueBox)
		.AddGroup(B_HORIZONTAL, 0)
			.Add(fRemoveButton)
			.Add(fAddButton)
			.End()
		.End();
```

Comment (owenca) [inline sources/ActionView.cpp]: Nice fix!

**Comment by owenca** [comment]
tight

**Comment by humdingerb** [comment]
Thanks!

**Pull request merged**